internal/worker/server.go

```diff
@@ -124,6 +112,31 @@ func (s *Server) VerifyIdentityHeader(nextHandler echo.HandlerFunc) echo.Handler
 	}
 }
 
+// This function should be started as a goroutine
+// Every 30 seconds it goes through all running jobs, removing any unresponsive ones.
+// It fails jobs which fail to check if they cancelled for more than 2 minutes.
+func (s *Server) WatchHeartbeats() {
+	for range time.Tick(time.Second * 30) {
+		for _, token := range s.jobs.Heartbeats(time.Second * 120) {
+			go func(token uuid.UUID) {
```

Comment:
Why is this running in a separate go routine?

Reply:
Well this should run throughout the lifetime of the server, as there isn't really a prompt to make it check for heartbeats.

Reply:
Yeah, I understand this, this is the reason why the whole `WatchHeartbeats` method runs in its own goroutine. I wonder why we spawn a separate one for each timed-out job. Can we just have one for everything?

Reply:
Oh the inner one. Yea there's no reason for that to be there, I'll remove it :D Thanks for noticing